lib/update/cluster/executor.go

```diff
@@ -86,57 +89,74 @@ func fsmSpec(c FSMConfig) fsm.FSMSpecFunc {
 			return nil, trace.BadParameter("unsupported operation %q", p.Plan.OperationType)
 		}
 
+		logger := &fsm.Logger{
+			FieldLogger: log.WithFields(log.Fields{
+				constants.FieldPhase: p.Phase.ID,
+			}),
+			Key:      fsm.OperationKey(p.Plan),
+			Operator: c.Operator,
+		}
+		if p.Phase.Data != nil {
+			logger.Server = p.Phase.Data.Server
+		}
+
 		switch p.Phase.Executor {
 		case updateInit:
-			return NewUpdatePhaseInit(c, p.Plan, p.Phase)
+			return libphase.NewUpdatePhaseInit(p, c.Operator, c.Apps,
+				c.Backend, c.LocalBackend, c.ClusterPackages, c.Users,
+				logger)
 		case updateChecks:
-			return NewUpdatePhaseChecks(c, p.Plan, p.Phase, c.Remote)
+			return libphase.NewUpdatePhaseChecks(p, c.Operator, c.Apps, c.Runner, logger)
 		case updateBootstrap:
-			return NewUpdatePhaseBootstrap(c, p.Plan, p.Phase, remote)
+			return libphase.NewUpdatePhaseBootstrap(p,
+				c.Operator, c.Apps,
```

Comment:
I had to go full dependency injection as the phases have been moved to a separate subpackage.